src/Egret.Cli/Formatters/AudacityResultFormatter.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace Egret.Cli.Formatters
+{
+    using Extensions;
+    using Models.Audacity;
+    using Models.Results;
+    using Processing;
+    using Serialization.Audacity;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Result formatter that stores output in Audacity (2.x) project files.
+    /// </summary>
+    public class AudacityResultFormatter : IResultFormatter
+    {
+        private readonly OutputFile outputFile;
+        private readonly AudacitySerializer serializer;
+        private readonly List<FileInfo> outputFiles;
+
+        public IReadOnlyCollection<FileInfo> OutputFiles => new ReadOnlyCollection<FileInfo>(outputFiles);
+
+        public AudacityResultFormatter(OutputFile outputFile, AudacitySerializer serializer)
+        {
+            this.outputFile = outputFile;
+            this.serializer = serializer;
+            this.outputFiles = new List<FileInfo>();
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            // no op
+            return ValueTask.CompletedTask;
+        }
+
+        public ValueTask WriteResultsHeader()
+        {
+            // no op
+            return ValueTask.CompletedTask;
+        }
+
+        public ValueTask WriteResult(int index, TestCaseResult result)
+        {
+            (FileInfo currentFile, Project project) = this.BuildProject(index, result);
+            this.serializer.Serialize(currentFile.FullName, project);
+
+            return ValueTask.CompletedTask;
+        }
+
+        public ValueTask WriteResultsFooter(FinalResults finalResults)
+        {
+            // TODO: what to do with the stats?
+            // for now, just ignore the footer data
+            return ValueTask.CompletedTask;
+        }
+
+        private (FileInfo, Project) BuildProject(int index, TestCaseResult result)
+        {
+            var sourceFile = new FileInfo(result.Context.SourceName);
+            var sourceName = sourceFile.Filestem();
+
+            var outputBaseFile = outputFile.GetOutputFile(".aup");
+            var outputName = outputBaseFile.Filestem();
+
+            var outputFullName = outputName + "_" + sourceName;
+            var outputFullFile = new FileInfo(Path.Combine(outputBaseFile.DirectoryName, outputFullName + ".aup"));
```

Comment:
Add an overload to `GetOutputFile` that can accept tags to add to the filename

```suggestion
            var sourceName = Path.GetFileNameWithoutExtension(result.Context.SourceName);

            var outputFile = outputFile.GetOutputFile(".aup", sourceName);
```